algorithms/src/Kokkos_Sort.hpp

```diff
@@ -648,8 +656,11 @@ sort(const ExecutionSpace& exec,
 template <class DataType, class... Properties>
 void sort(const Experimental::SYCL& space,
           const Kokkos::View<DataType, Properties...>& view) {
-  using ViewType = Kokkos::View<DataType, Properties...>;
+  if (view.extent(0) == 0) {
+    return;
+  }
```

Comment:
you mean this is not necessary because it just happens that oneapi handles that case anyway? 
I think it is important to not rely on assumptions about what happens impl-wise. 
basically, what i am saying is that for constraints and preconditons and corner cases, each overload of a function should be see at a high-level as:
```cpp
// constraints 
void function(...){. //overload 1
 // preconditions 
 // corner cases
 // impl call 
}

// constraints 
void function(...){. //overload 2
 // preconditions 
 // corner cases
 // impl call 
}
```
and so just because overload2 happens to call overload1 or something else, the preconditions or corner cases should be explicitly done in each. Suppose that we did not do this but just rely on the fact that the corner case is handled in overload1. Then someone later changes the impl and forgets about the corner case. This messes everything up.

Reply:
@dalg24 makes sense?

Reply:
I would be comfortable with deferring to Thrust/oneDPL/the standard C++ library for something that trivial.
It is less code to maintain on our side.  That said, your approach is fine.